public/app/features/alerting/routes.tsx

```diff
@@ -137,35 +149,55 @@ const unifiedRoutes: RouteDescriptor[] = [
   },
   {
     path: '/alerting/notifications',
-    roles: config.unifiedAlertingEnabled ? () => ['Editor', 'Admin'] : undefined,
+    roles: () =>
+      contextSrv.evaluatePermission(config.unifiedAlertingEnabled ? () => ['Editor', 'Admin'] : () => [], [
+        AccessControlAction.AlertingNotificationsRead,
+        AccessControlAction.AlertingNotificationsExternalRead,
+      ]),
     component: SafeDynamicImport(
       () => import(/* webpackChunkName: "NotificationsListPage" */ 'app/features/alerting/unified/Receivers')
     ),
   },
   {
     path: '/alerting/notifications/templates/new',
-    roles: () => ['Editor', 'Admin'],
+    roles: () =>
+      contextSrv.evaluatePermission(config.unifiedAlertingEnabled ? () => ['Editor', 'Admin'] : () => [], [
+        AccessControlAction.AlertingNotificationsCreate,
+        AccessControlAction.AlertingNotificationsExternalWrite,
+      ]),
     component: SafeDynamicImport(
       () => import(/* webpackChunkName: "NotificationsListPage" */ 'app/features/alerting/unified/Receivers')
     ),
   },
   {
     path: '/alerting/notifications/templates/:id/edit',
-    roles: () => ['Editor', 'Admin'],
+    roles: () =>
+      contextSrv.evaluatePermission(config.unifiedAlertingEnabled ? () => ['Editor', 'Admin'] : () => [], [
+        AccessControlAction.AlertingNotificationsUpdate,
+        AccessControlAction.AlertingNotificationsExternalWrite,
+      ]),
     component: SafeDynamicImport(
       () => import(/* webpackChunkName: "NotificationsListPage" */ 'app/features/alerting/unified/Receivers')
     ),
   },
   {
     path: '/alerting/notifications/receivers/new',
-    roles: () => ['Editor', 'Admin'],
+    roles: () =>
+      contextSrv.evaluatePermission(config.unifiedAlertingEnabled ? () => ['Editor', 'Admin'] : () => [], [
+        AccessControlAction.AlertingNotificationsCreate,
+        AccessControlAction.AlertingNotificationsExternalWrite,
+      ]),
     component: SafeDynamicImport(
       () => import(/* webpackChunkName: "NotificationsListPage" */ 'app/features/alerting/unified/Receivers')
     ),
   },
   {
     path: '/alerting/notifications/receivers/:id/edit',
-    roles: () => ['Editor', 'Admin'],
+    roles: () =>
```

Comment:
Maybe we could extract all of these functions to a single function? Like `evaluateNotificationsAccessRoles` or something similar?